compiler/scanner.mll

```diff
@@ -0,0 +1,75 @@
+(*
+ * Odds scanner
+ *
+ * Authors: Alex Kalicki, Alexandra Medway, Daniel Echikson, Lilly Wang
+ * COMS4115
+ *)
+
+{ open Parser }
+
+let num = ['0'-'9']
+
+rule token = parse
+  
+(* Whitespace *)
+| [' ' '\n' '\r'] { token lexbuf }
+
+(* Comments *)
+| "/*"		{ comment lexbuf }
+
+(* Punctuation *)
+| '('		{ LPAREN }  | ')'		{ RPAREN }
+| '<'		{ LCAR }	| '>'		{ RCAR } (* Also relational operators *)
+| '['		{ LBRACK }	| ']'		{ RBRACK }
+| ';'		{ SEMI }    | ':'		{ COLON }
+| ','		{ COMMA }	| '|'		{ VBAR }
+
+(* Arithmetic Operators *)
+| '+'		{ PLUS }    | '-'		{ MINUS }
+| '*'		{ TIMES }   | '/'		{ DIVIDE }
+| '%'		{ MOD }
+
+(* Relational Operators *)
+| "=="		{ EQ }		| "!="		{ NEQ }
+| "<="		{ LEQ }		| ">="		{ GEQ }
+
+(* Logical Operators & Keywords*)
+| "&&"		{ AND }		| "||"		{ OR }
+| "!"		{ NOT }
+
+(* Assignment Operator *)
+| '='		{ ASN }
+
+(* Conditional Operators *)
+| "if"		{ IF }		| "then"	{ THEN }
+| "else"	{ ELSE }
+
+(* Declarative Keywords *)
+| "set"		{ SET }		| "state"	{ STATE }
+
+(* Function Symbols & Keywords *)
+| "->"		{ FDELIM }	| "=>"		{ FRTYPE }
+| "return"	{ RETURN }
+
+(* Type Keywords *)
+| "int"		{ INT }		| "float"	{ FLOAT }
+| "string"	{ STRING }	| "dist"	{ DIST }
+| "list"	{ LIST }	| "void"	{ VOID }
+| "bool"	{ BOOL }
+
+(* End-of-File *)
+| eof { EOF }
+
+(* Identifiers *)
+| ['a'-'z' 'A'-'Z'] ['a'-'z' 'A'-'Z' '_']* as lxm { ID(lxm) }
```

Comment:
We want to be able to do variables starting with an underscore and having numbers in the second half. So maybe

```ocaml
| ['a'-'z' 'A'-'Z' '_'] ['a'-'z' 'A'-'Z' '_' '0'-'9']* as lxm { ID(lxm) }
```